File: Serenity.Script.UI/Editor/PasswordEditor.cs
using jQueryApi;
using System;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serenity
{
    [Imported(ObeysTypeSystem = true), Editor, DisplayName("Şifre")]
    public class PasswordEditor : StringEditor
    {
        public PasswordEditor(jQueryObject input)
            : base(input)
        {
            input.Attribute("type", "password");
        }
    }
}

File: src/Serenity.Scripts/CoreLib/UI/Editors/PasswordEditor.ts
import { registerEditor } from "../../Decorators";
import { StringEditor } from "./StringEditor";

@registerEditor('Serenity.PasswordEditor')
export class PasswordEditor extends StringEditor {
    constructor(input: JQuery) {
        super(input);

        input.attr('type', 'password');
    }
}